client/src/components/history/history-chat.tsx

```diff
@@ -0,0 +1,169 @@
+
+interface ScrollingQuestionsParams {
+  height: number;
+  questionHistory: string[];
+}
+
+const MENTOR_COLORS = ["#d8e7f8", "#d4e8d9", "#ffebcf", "#f5cccd"];
+
+export function HistoryChat(args: ScrollingQuestionsParams): JSX.Element {
+  const { height } = args;
+  const styles = useStyles();
+
+  const colorByMentorId = useSelector<State, Record<string, string>>((s) => {
+    const mentorIds = Object.getOwnPropertyNames(s.mentorsById);
+    mentorIds.sort();
+    return mentorIds.reduce<Record<string, string>>((acc, cur, i) => {
+      acc[cur] = MENTOR_COLORS[i % MENTOR_COLORS.length];
+      return acc;
+    }, {});
+  });
+  const namesByMentorId = useSelector<State, Record<string, string>>((s) => {
+    const mentorIds = Object.getOwnPropertyNames(s.mentorsById);
+    mentorIds.sort();
+    return mentorIds.reduce<Record<string, string>>((acc, cur) => {
+      acc[cur] = s.mentorsById[cur].mentor.name;
+      return acc;
+    }, {});
+  });
+
+  const chatData = useSelector<State, ChatData>((s) => s.chat);
+
+  const dispatch = useDispatch();
+  const [checked, toggleChecked] = useState<boolean>(true);
+
+  useEffect(() => {
+    animateScroll.scrollToBottom({
+      containerId: "chat-thread",
+    });
+  }, [chatData.messages]);
+
+  const toggleAnswers = (
+    <div>
+      <FormGroup className="togglePos">
+        <FormControlLabel
+          control={
+            <Switch
+              size="medium"
+              checked={checked}
+              onChange={() => toggleChecked((prev) => !prev)}
+              data-cy="visibility-switch"
+            />
+          }
+          label="show/hide answers"
+        />
+      </FormGroup>
+    </div>
+  );
+
+  useEffect(() => {
+    dispatch(visibilitySwitch(chatData.messages.length, checked));
+  }, [checked]);
+
+  const totalMentors: Array<string> = [
+    ...new Set(chatData.messages.map((item) => item.mentorId)),
+  ];
+
+  return (
+    <div
+      data-cy="history-chat"
+      data-topic="History"
+      className={styles.chat_container}
+    >
+      <List
+        data-cy="chat-thread"
+        className={styles.list}
+        style={{ height: height }}
+        disablePadding={true}
+        id="chat-thread"
+      >
+        {toggleAnswers}
+        {chatData.messages.map((m: ChatMsg, i: number) => {
+          const itemData: ChatItemData = {
+            ...m,
+            color: colorByMentorId[m.mentorId] || "",
+            name: namesByMentorId[m.mentorId] || "",
+            isVisible: m.visibility,
+          };
+          return (
+            <div
+              key={i}
+              style={{
+                display: "flex",
+                flexDirection: "row",
+                alignItems: "center",
+              }}
+            >
+              <ChatItem
+                key={`chat-msg-${i}`}
+                message={itemData}
+                i={i}
+                styles={styles}
+                totalMentors={totalMentors.filter((msg) => msg).length}
```

Comment:
we just want a count of mentors excluding the user? Can this not just be

```typescript
totalMentors={Object.getOwnPropertyNames(namesByMentorId).length}
```

?

Reply:
Yes, I already changed it